samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,67 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %s\n", SDL_IsGameController(gameControllerIndex)?"Yes":"No");
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("\n");
+
+  while (true) {
+    // Fetch current GameController state
+    SDL_GameControllerUpdate();
+    debugClearScreen();
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_A),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_A));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_B),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_B));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_X),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_X));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_Y),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_Y));
+    debugPrint("%s:%i                 ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_BACK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_BACK));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_START),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_START));
+    debugPrint("%s:%i       ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_LEFTSTICK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_LEFTSTICK));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_RIGHTSTICK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_RIGHTSTICK));
+    debugPrint("%s:%i ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_LEFTSHOULDER),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_LEFTSHOULDER));
```

Comment:
Are these using the correct "Black" / "White" names?

These should map as they do for backwards compatibility on 360 (official MS mapping).

There should probably be a comment in the source code which explains this detail.

Reply:
They output the x360 mapping name (LeftTrigger and RightTrigger [or something like that]).
Mapping is 1:1 what was used by XInput in SDL_gamecontrollerdb.h, I'm not sure how to map it better.

Reply:
These names must be "White" and "Black" for original Xbox controllers, whereas they would probably be something like "LB" (Left-Button) and "RB" (Right-Button) [or the other way around] on 360 controllers.

Our gamecontroller mapping has to reflect that.

Reply:
I have not seen any indication that button names can be remapped freely.
I believe the whole idea of the GameController API is to make sure that the button SDL_GAMECONTROLLER_BUTTON_A always is the one located closest to where A is located on an x360 controller and that the string returned is always "a" for that particular `enum` value? I might be wrong though, the documentation of the string getter is not very clear.

Reply:
Looks like labels are not implemented yet for gamecontroller mappings.

[This code in SDL2](https://github.com/spurious/SDL-mirror/blob/74e23455ab2abbced7d1ef6c610449129582788f/src/joystick/SDL_gamecontroller.c#L492) even suggests that these names are lowercase unique identifiers (key-like).

In that case the code can remain as-is. I was under the impression gamecontroller-labels had existed in SDL2 already.